### DISCUSSION TITLE
Searching for time in the last 90 days

### DISCUSSION THREAD

**Discussion by hippyjm** [SigmaHQ/sigma#4599] [Q&A] [answered]
Hopefully a simple one. Im looking to only get results if an event occurred in the last 90 days to get rid of a lot of noise. im unsure of the syntax / commands i require and cant seem to find any information online. Ive tried time-window. timeframe doesn't work as that's just not right :) 
basically i want to see if that event has occoured in the last 90 days

`logsource:
    product: windows
    service: system
detection:
    selection:
        EventID:
            - 16384
    condition: selection`

**Comment by phantinuss** [comment]
This is something you need to define in your post-processing pipeline in pySigma for your specific SIEM use case. There's an example on how to do that for Splunk in this blog post: https://blog.sigmahq.io/introducing-query-post-processing-and-output-finalization-to-processing-pipelines-4bfe74087ac1

I am not aware of other public documentation about this, right now.

**Reply by hippyjm** [reply]
thanks for responding. no worries. trying to think of other ways of getting round the noise for that one event.
Any ideas of, for instance selecting only the last 10 items of that rule. like in powershell you can do select -last10. 
just getting into sigma and its so powerful. just a steep learning curve.

**Reply by phantinuss** [reply]
Sigma is an intermediate language mainly for knowledge transfer. What you are asking about is on how to implement something for a specific ?SIEM?. I don't know. Without more explanation on what exactly you are working on and with which tools, there's no way to help.

**Reply by hippyjm** [reply]
Using it with chainsaw.exe to look at event logs and determine any issues. the issue is that some rules are very repetative. we do care about them but hoping to streamline it a bit as it can take a while to parse the logs. all good though and really enjoying the work

**Reply by phantinuss** [reply]
Yeah, so it's more a question on how to use chainsaw. Better ask there if there are workflows that help your use case.

**Reply by hippyjm** [reply, marked as answer]
Will do. ive found a bit of a messy way to get round the issue
detection:
    selection:
        EventID:
            - 16384
        Event.System.TimeCreated:
            - '2023-07*'
            - '2023-08*'
            - '2023-09*'
            - '2023-10*'
            - '2023-11*'
            - '2023-12*'
            - '2024-*'
    condition: selection
            
    as i want logs further back for everything else. just not this one. ill need to maintain and keep on top of it but hey ho. the fun of IT
